clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -761,48 +762,79 @@ static CompoundStmt *CreateOpenCLKernelBody(Sema &S,
+                                           WrapperFldCRD, Base, InitMethodName,
+                                           BodyStmts);
+                } else if (Util::isSyclSpecConstantType(FldType)) {
+                  // Specialization constants are "invisible" to the
+                  // kernel argument creation and device-side SYCL object
+                  // materialization infrastructure in this source.
+                  // It is OK not to really materialize them on the kernel
+                  // side, because their only use can be via
+                  // 'spec_const_obj.get()' method, which is translated to
+                  // an intrinsic and 'this' is really never used.
+                } else {
+                  // Field is a structure or class so change the wrapper
+                  // object and recursively search for accessor field.
+                  DeclAccessPair WrapperFieldDAP =
+                      DeclAccessPair::make(WrapperFld, AS_none);
+                  auto NewBase = MemberExpr::Create(
+                      S.Context, Base, false, SourceLocation(),
+                      NestedNameSpecifierLoc(), SourceLocation(), WrapperFld,
+                      WrapperFieldDAP,
+                      DeclarationNameInfo(WrapperFld->getDeclName(),
+                                          SourceLocation()),
+                      nullptr, WrapperFld->getType(), VK_LValue, OK_Ordinary,
+                      NOUR_None);
+                  getExprForWrappedAccessorInit(WrapperFldCRD, NewBase);
+                }
+              } else if (FldType->isConstantArrayType()) {
+                const ConstantArrayType *CAT = cast<ConstantArrayType>(FldType);
+                QualType ET = CAT->getElementType();
+                if (Util::isSyclAccessorType(ET)) {
+                  size_t NumElements =
+                      static_cast<size_t>(CAT->getSize().getZExtValue());
+                  // Treat each element of accessor array as a separate item
+                  // Call its __init function
+                  for (size_t I = 0; I < NumElements; ++I) {
+                    FieldDecl *Field = WrapperFld;
+                    DeclAccessPair FieldDAP =
+                        DeclAccessPair::make(Field, AS_none);
+                    auto SpecialObjME = MemberExpr::Create(
+                        S.Context, Base, false, SourceLocation(),
+                        NestedNameSpecifierLoc(), SourceLocation(), Field,
+                        FieldDAP,
+                        DeclarationNameInfo(Field->getDeclName(),
+                                            SourceLocation()),
+                        nullptr, Field->getType(), VK_LValue, OK_Ordinary,
+                        NOUR_None);
+                    ExprResult IndexExpr =
+                        S.ActOnIntegerConstant(SourceLocation(), I);
+                    auto MA = S.CreateBuiltinArraySubscriptExpr(
```

Comment:
Please do not use auto jere.

Reply:
All changes suggested by you have been made.

Reply:
Thanks!